blaze-client/src/test/scala/org/http4s/client/blaze/Http1ClientStageSuite.scala

```diff
@@ -0,0 +1,296 @@
+  }
+
+  test("Utilize a provided Host header") {
+    val resp = "HTTP/1.1 200 OK\r\n\r\ndone"
+
+    val req = FooRequest.withHeaders(headers.Host("bar.test"))
+
+    getSubmission(req, resp).map { case (request, response) =>
+      val requestLines = request.split("\r\n").toList
+      assert(requestLines.contains("Host: bar.test"))
+      assertEquals(response, "done")
+    }
+  }
+
+  test("Insert a User-Agent header") {
+    val resp = "HTTP/1.1 200 OK\r\n\r\ndone"
+
+    getSubmission(FooRequest, resp, DefaultUserAgent).map { case (request, response) =>
+      val requestLines = request.split("\r\n").toList
+      assert(requestLines.contains(s"User-Agent: http4s-blaze/${BuildInfo.version}"))
+      assertEquals(response, "done")
+    }
+  }
+
+  test("Use User-Agent header provided in Request") {
+    import org.http4s.syntax.all._
+
+    val resp = "HTTP/1.1 200 OK\r\n\r\ndone"
+    val req = FooRequest.withHeaders(Header.Raw("User-Agent".ci, "myagent"))
+
+    getSubmission(req, resp).map { case (request, response) =>
+      val requestLines = request.split("\r\n").toList
+      assert(requestLines.contains("User-Agent: myagent"))
+      assertEquals(response, "done")
+    }
+  }
+
+  fooConnection.test("Not add a User-Agent header when configured with None") { tail =>
+    val resp = "HTTP/1.1 200 OK\r\n\r\ndone"
+
+    getSubmission(FooRequest, resp, tail).map { case (request, response) =>
+      val requestLines = request.split("\r\n").toList
+      assertEquals(requestLines.find(_.startsWith("User-Agent")), None)
+      assertEquals(response, "done")
+    }
+  }
+
+  // TODO fs2 port - Currently is elevating the http version to 1.1 causing this test to fail
+  test("Allow an HTTP/1.0 request without a Host header".ignore) {
+    val resp = "HTTP/1.0 200 OK\r\n\r\ndone"
+
+    val req = Request[IO](uri = www_foo_test, httpVersion = HttpVersion.`HTTP/1.0`)
+
+    getSubmission(req, resp).map { case (request, response) =>
+      assert(!request.contains("Host:"))
+      assertEquals(response, "done")
+    }
+  }
+
+  test("Support flushing the prelude") {
+    val req = Request[IO](uri = www_foo_test, httpVersion = HttpVersion.`HTTP/1.0`)
+    /*
+     * We flush the prelude first to test connection liveness in pooled
+     * scenarios before we consume the body.  Make sure we can handle
+     * it.  Ensure that we still get a well-formed response.
+     */
+    getSubmission(req, resp).map(_._2).assertEquals("done")
+  }
+
+  fooConnection.test("Not expect body if request was a HEAD request") { tail =>
+    val contentLength = 12345L
+    val resp = s"HTTP/1.1 200 OK\r\nContent-Length: $contentLength\r\n\r\n"
+    val headRequest = FooRequest.withMethod(Method.HEAD)
+
+    val h = new SeqTestHead(List(mkBuffer(resp)))
+    LeafBuilder(tail).base(h)
+
+    tail.runRequest(headRequest, IO.never).flatMap { response =>
+      assertEquals(response.contentLength, Some(contentLength))
+
```

Comment:
The crucial change is here and might not be obvious in the merges. We deleted the assertion that the connection is immediately `isRecyclable`.